integration/alertmanager_test.go

```diff
@@ -79,7 +80,8 @@ func TestAlertmanagerStoreAPI(t *testing.T) {
 	err = c.SetAlertmanagerConfig(context.Background(), cortexAlertmanagerUserConfigYaml, map[string]string{})
 	require.NoError(t, err)
 
-	require.NoError(t, am.WaitSumMetrics(e2e.Equals(1), "cortex_alertmanager_configs"))
+	time.Sleep(2 * time.Second)
```

Comment:
I solved the CI failure using a quick hack, but I dislike `Sleep` is tests as it a general source of flaky tests. The problem here is the metric does not get initialise registered (given it is per tenant) until much later. 

A solution I can see would be to create a new method (or argument to `WaitSumMetrics`) that waits for a metric to appear instead of bailing out if we don't see the metric on the initial metrics page load. As I can see this being useful in other areas. 

Please let me know and I'll be happy to make the change.

Reply:
Let's work on it in a separate PR. Could you open an issue, please? I have this pending PR https://github.com/cortexproject/cortex/pull/2522 which could be used as a baseline to add a functional option to enable the logic you suggest.

Reply:
Thanks marco, filed in https://github.com/cortexproject/cortex/issues/2975